src/rest-server/src/controllers/v2/user.js

```diff
@@ -103,7 +103,15 @@ const createUserIfUserNotExist = async (req, res, next) => {
       grouplist = await groupModel.getUserGrouplistFromExternal(username, data);
       req.grouplist = grouplist;
       if (grouplist && grouplist.length === 0) {
-        return next(createError('Forbidden', 'ForbiddenUserError', `User ${userData.username} is not in configured groups.`));
+        let forbiddenMessage = `User ${userData.username} is not in configured groups.`;
+        if (authConfig.groupConfig.groupDataSource === 'ms-graph') {
+          forbiddenMessage = forbiddenMessage + `Please contact your admin, and join the AAD group named [ ${authConfig.groupConfig.defaultGroup.externalName} ].`;
+        }
+        return res.redirect(req.returnBackURI + '?'+ querystring.stringify({
+          from: fromURI,
+          loginFailMsg: forbiddenMessage
+        }));
```

Comment:
should throw error here and handle all kinds of errors in [`getAAD`](https://github.com/microsoft/pai/blob/master/src/rest-server/src/controllers/v2/token.js#L76)

Reply:
`from` field seems useless here because it's better to redirect user to index page instead of `fromURI`?

Reply:
It seems redirecting user to the origin page will be better？

Reply:
fixed